### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by mattwiller**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @mattwiller has signed the CLA. Thanks for the pull request!

File: boxsdk/client/client.py [hunk 1]
```diff
@@ -505,6 +505,7 @@ def get_groups(self, name=None, limit=None, offset=None, fields=None):
         return LimitOffsetBasedObjectCollection(
             url=url,
             session=self._session,
+            additional_params=additional_params,
```
Comment (Jeff-Meadows) [inline boxsdk/client/client.py]: this doesn't seem to fit with the rest of the PR

File: boxsdk/object/base_object.py [hunk 2]
```diff
@@ -169,63 +169,6 @@ def __ne__(self, other):
     def __hash__(self):
         return hash((self._object_id, self._item_type))
 
-    def _paging_wrapper(self, url, starting_index, limit, factory=None):
```
Comment (Jeff-Meadows) [inline boxsdk/object/base_object.py]: I guess removing unused code will increase the test coverage (by percentage of lines covered), but this seems like it could be a separate commit

File: boxsdk/client/client.py (hunk 1 above)
Comment (mattwiller) [reply boxsdk/client/client.py]: It's a bugfix, discovered while adding test coverage for this method.

File: test/unit/object/test_legal_hold_policy.py [hunk 3]
```diff
@@ -119,20 +119,32 @@ def test_delete_policy_return_the_correct_response(
...
+])
 def test_get_assignments(
         test_legal_hold_policy,
         mock_box_session,
         policies_response,
         policy_id_1,
         policy_id_2,
+        assign_to_type,
+        assign_to_id,
+        params
```
Comment (Jeff-Meadows) [inline test/unit/object/test_legal_hold_policy.py]: trailing comma

File: test/unit/object/test_legal_hold_policy.py [hunk 4]
```diff
@@ -119,20 +119,32 @@ def test_delete_policy_return_the_correct_response(
     assert response is True
 
 
+@pytest.mark.parametrize('assign_to_type, assign_to_id, params', [
+    (None, None, {}),
+    ('file', None, {'assign_to_type': 'file'}),
+    ('folder', '22222', {'assign_to_type': 'folder', 'assign_to_id': '22222'})
```
Comment (Jeff-Meadows) [inline test/unit/object/test_legal_hold_policy.py]: In all 3 cases, params is duplicate information. Is it worth having it as a separate parameter to the test?

File: test/unit/object/test_item.py [hunk 5]
```diff
@@ -320,15 +330,25 @@ def test_collaborate_with_user(test_item_and_response, mock_user, mock_box_sessi
...
-def test_collaborate_with_login(test_item_and_response, mock_box_session):
+@pytest.mark.parametrize('can_view_path,fields,notify,data,params', [
+    (None, None, None, {}, {}),
+    (True, None, None, {'can_view_path': True}, {}),
+    (False, None, None, {'can_view_path': False}, {}),
+    (None, ['type', 'id', 'created_by'], None, {}, {'fields': 'type,id,created_by'}),
+    (None, None, True, {}, {'notify': True}),
+    (None, None, False, {}, {'notify': False}),
+    (True, ['type', 'id', 'created_by'], False, {'can_view_path': True}, {'fields': 'type,id,created_by', 'notify': False})
+])
```
Comment (Jeff-Meadows) [inline test/unit/object/test_item.py]: Is this the exact same params as for `test_collaborate_with_user`? May want to DRY this up.
